OpenSim/Simulation/Model/MovingPathPoint.cpp

```diff
@@ -72,364 +63,211 @@ MovingPathPoint::~MovingPathPoint()
+        if (yCoord != aNode.element_end())
+            yCoord->getValueAs<std::string>(yCoord_name);
+        if (zCoord != aNode.element_end())
+            zCoord->getValueAs<std::string>(zCoord_name);
+        XMLDocument::addConnector(aNode, "Connector_Coordinate_", 
+            "x_coordinate", xCoord_name);
+        XMLDocument::addConnector(aNode, "Connector_Coordinate_", 
+            "y_coordinate", yCoord_name);
+        XMLDocument::addConnector(aNode, "Connector_Coordinate_", 
+            "z_coordinate", zCoord_name);
     }
-}
 
-//=============================================================================
-// OPERATORS
-//=============================================================================
-//_____________________________________________________________________________
-/**
- * Assignment operator.
- *
- * @return Reference to this object.
- */
-MovingPathPoint& MovingPathPoint::operator=(const MovingPathPoint &aPoint)
-{
-    // BASE CLASS
-    PathPoint::operator=(aPoint);
-
-    copyData(aPoint);
-
-    return(*this);
+    // Call base class now assuming _node has been corrected for current version
+    Super::updateFromXMLNode(aNode, versionNumber);
 }
 
-//_____________________________________________________________________________
-/**
- * Update the point's location.
- *
- */
-void MovingPathPoint::update(const SimTK::State& s)
+SimTK::Vec3 MovingPathPoint::getLocation(const SimTK::State& s) const
 {
-    if (_xCoordinate) {
+    SimTK::Vec3 pInF(0);
+    if (!_xCoordinate.empty()) {
         const double xval = SimTK::clamp(_xCoordinate->getRangeMin(),
-                                         _xCoordinate->getValue(s),
-                                         _xCoordinate->getRangeMax());
-        upd_location()[0] = _xLocation->calcValue(SimTK::Vector(1, xval));
-    } else // type == Constant
-        upd_location()[0] = _xLocation->calcValue(SimTK::Vector(1, 0.0));
+            _xCoordinate->getValue(s),
+            _xCoordinate->getRangeMax());
+        pInF[0] = get_x_location().calcValue(SimTK::Vector(1, xval));
+    }
+    else // assume a Constant
+        pInF[0] = get_x_location().calcValue(SimTK::Vector(1, 0.0));
 
-    if (_yCoordinate) {
+    if (!_yCoordinate.empty()) {
         const double yval = SimTK::clamp(_yCoordinate->getRangeMin(),
-                                         _yCoordinate->getValue(s),
-                                         _yCoordinate->getRangeMax());
-        upd_location()[1] = _yLocation->calcValue(SimTK::Vector(1, yval));
-    } else // type == Constant
-        upd_location()[1] = _yLocation->calcValue(SimTK::Vector(1, 0.0));
+            _yCoordinate->getValue(s),
+            _yCoordinate->getRangeMax());
+        pInF[1] = get_y_location().calcValue(SimTK::Vector(1, yval));
+    }
+    else // type == Constant
+        pInF[1] = get_y_location().calcValue(SimTK::Vector(1, 0.0));
 
     if (_zCoordinate) {
```

Comment:
`_zCoordinate` -> `!_zCoordinate.empty()` to agree with changes to x and y cases?